### ISSUE TITLE
Default branch not recognised with Gogs

### ISSUE THREAD

**Issue by pokrakam** [abapGit/abapGit#4703]
Using [Gogs](https://gogs.io), here seems to be an issue that abapGit doesn’t recognise the default branch. When comparing a debug of Github vs Gogs, the branch list parser in `zcl_abapgit_git_branch_list` finds a name `HEAD` from Github where Gogs sends `refs/heads/main`. Some of the code seems to look for this pattern to put it into `head_symref`, but it doesn't find the Gogs variant. 

I can't list the branches on Gogs, but I can switch to one so it's not a showstopper. However it looks like something that should fixable. Will have a more detailed look when I find more time.

**Comment by larshp** [comment]
can the error be reproduced on the gogs demo platform?

**Comment by pokrakam** [comment]
Hmm, having a spot of bother attaching to it as my NPL system doesn't want to connect via SSL, even after importing all possible PEMs into STRUST. Have created an account and test repo on https://try.gogs.io/ user abapGit, pw via Slack

**Comment by pokrakam** [comment]
From what I can see this is the bit that could be changed, but am not 100% certain about side-effects. In the case of Gogs, the lv_name already contains the head_symref, and the parse_head_params doesn't find it because it's not in the parameters.
https://github.com/abapGit/abapGit/blob/f18a8ee82b693513b9c0cb7e24d5f5cc6676acd2/src/git/zcl_abapgit_git_branch_list.clas.abap#L269-L275

**Comment by pokrakam** [comment]
OK, I figured the cause is possibly a small bug in the way I created the Gogs repo. But I can also reproduce it by putting the remote repo in a detached head state, which I'm guessing may not be specific to Gogs.

In detail: I created a bare repo on the server which defaults to a branch `master`. I created a local repo, set the remote and pushed with a new branch `main`. 
If I look in the server repo, I can see `refs/heads` only contains one entry `main` - as expected. But `HEAD` still refers to `/refs/heads/master`. I may raise this on Gogs (a really good local Git Server BTW).

Now Gogs kindof does the right thing here when abapGit tries to pull, and won't send `HEAD` (presumably because it's invalid), and sends the `refs/heads/main` as the object name instead.  I can't `git clone` it in this state either.

So the big question is whether we should cater for this or this is just too obscure? A simple hack would be if we can't find a symref then see if we find a `lv_name` starting with `refs/heads/` and use that as a default branch. Is it worthwhile or opens up too much guesswork?
